packages/flutter/test/widgets/async_test.dart

```diff
@@ -87,6 +88,20 @@ void main() {
     });
   });
   group('FutureBuilder', () {
+    testWidgets('gives expected snapshot with SynchronousFuture', (WidgetTester tester) async {
+      final SynchronousFuture<String> future = SynchronousFuture<String>('flutter');
+      await tester.pumpWidget(FutureBuilder<String>(
+        future: future,
+        builder: (BuildContext context, AsyncSnapshot<String> snapshot) {
+          expect(snapshot.connectionState, ConnectionState.done);
```

Comment:
Before this patch, this expectation failed (it would say ConnectionState.waiting and never get called back again because the `then` block got called synchronously).